packages/web/src/shared/components/AutocompleteInput.tsx

```diff
@@ -96,13 +100,35 @@ export default function AutocompleteInput({
     selectItem(items.find((item) => item.value === value) ?? null);
   }, [value, selectItem, items]);
 
+  // The width of textMeasureElement allows us to measure the width of the input text.
+  // That width is used to size the overall component.
+  const textMeasureElement = useRef(null);
+  const [measuredWidth, setMeasuredWidth] = useState(0);
+  function updateMeasuredWidth() {
+    if (textMeasureElement.current) {
+      setMeasuredWidth((textMeasureElement.current as HTMLElement).clientWidth);
+    }
+  }
+  useEffect(() => updateMeasuredWidth(), [textMeasureElement]);
```

Comment:
What do you think about pulling this out into its own custom hook? We could use it like this in TranslateWord.
```js
const [text, setText] = useState('asdf')
const { width } = useTextWidth(text)

return <AutocompleteInput value={text} style={{ width: width + 56 }} />
```

That hook can also manage its own element to use for measuring:
```js
const el = useRef<HTMLDivElement>()
useEffect(() => {
  const div = document.createElement('div')
  document.body.appendChild(div)
  el.current = div
  return () => div.remove()
}, [])

useEffect(() => {
  el.current.innerText = text
}, [text])
```